spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1574,7 +1579,7 @@ def set_language(self, language, filename=None):
                     self.comment_string = comment_string
                     if key in CELL_LANGUAGES:
                         self.supported_cell_language = True
-                        self.cell_separators = CELL_LANGUAGES[key]

```

Comment:
I do not understand why this change was needed @impact27

Reply:
We do not need to know the separators but just that is supports cells?

Reply:
Because the whole cell logic was separated between two systems: the syntax highlighter that keeps a list of cell for efficient cell finding, and the editor that reads multiple times through the whole file to find cells.

Reply:
So I removed the second system as it was causing performances issues on large files.

Reply:
Understood